:sparkles: Add nonbootable iso api

hroyrh opened the pull request:
**What this PR does / why we need it**:
This PR defines the API for attaching and detaching non-bootable iso image. The Controller implementation for the API is in progress in this pr : https://github.com/metal3-io/baremetal-operator/pull/1594

dtantsur commented on apis/metal3.io/v1alpha1/dataimage_types.go:
```diff
@@ -0,0 +1,82 @@
+
+// DataImageSpec defines the desired state of DataImage.
+type DataImageSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
```
nit: please remove

dtantsur commented on apis/metal3.io/v1alpha1/dataimage_types.go:
```diff
@@ -0,0 +1,82 @@
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:shortName=dimg
```
It's going to be a very rarely used resource, let's maybe not have a short name for it?

dtantsur commented:
The previous PR modifies a few more API-related files, such as config/base/crds/kustomization.yaml, a sample or RBAC roles. Could you add them too?

hroyrh commented:
@dtantsur I have updated the base kustomization config, and added cainjection, webhook and sample dataimage resource file. I've left the RBAC configs since they depend on the controller and it will be more appropriate to include them in the other pr.

elfosardo commented:
/test metal3-bmo-e2e-test-pull

elfosardo commented:
/test-centos-e2e-integration-main

dtantsur commented:
/approve

elfosardo commented:
/lgtm
